src/textual/app.py

```diff
@@ -2198,7 +2198,8 @@ async def prune_widgets_task(
                 await self._prune_nodes(widgets)
             finally:
                 finished_event.set()
-                self.refresh(layout=True)
+                if parent.styles.auto_dimensions:
+                    parent.refresh(layout=True)
```

Comment:
Don't we need to refresh the layout of the parent, regardless of whether they have auto dimensions or not?
Imagine the parent has fixed dimensions but the children have relative dimensions (e.g., `fr` units).
Don't we need the refresh to make sure the children make use of the space that was just made available?

Reply:
auto dimensions and relative units kind of cancel each other out.

Consider a container with auto height and a single child with height `1fr`. The widget needs to know the height of the container, but the container needs to know the height of the child. So auto height considers all relative dimensions to be zero. It will still respect minimums, so essentially what it calculates is the minimal value which satisfies all constraints.

Reply:
That's interesting.
Up until now, I interpreted `1fr` to mean that it would expand as much as possible while also respecting relative ratios to other widgets with `fr` dimensions.
Then, if the parent has `auto` and the child has `1fr`, the child should look at the grandparent to find the maximum size it can occupy and the parent would expand to that size.
This is the behaviour I find more consistent with the interaction between absolute sizes and `1fr`.
If a parent has a fixed size and the child has `1fr`, the child will _expand as much as possible_. Hence, I have been taking the "expand as much as possible" as the interpretation of `1fr`, which I don't think should fall apart when the parent is `auto`.